src/libs/antares/study/parts/short-term-storage/series.cpp

```diff
@@ -42,29 +44,75 @@ bool Series::validate() const
         logs.warning() << "Size of series for short term storage is wrong";
         return false;
     }
+
+    auto checkVectPositive = [](std::vector<double> v) {
+        return std::count_if(v.begin(), v.end(), [](double d){ return d >= 0.0; });
+    };
+    if (!checkVectPositive(maxInjectionModulation) || !checkVectPositive(maxWithdrawalModulation))
+    {
+        logs.warning() << "Values for PMAX series should be positive";
+        return false;
+    }
+
+    auto checkVectBetween = [](std::vector<double> v) {
+        return std::count_if(v.begin(), v.end(), [](double d){ return (d >= 0.0 && d <= 1.0); });
+    };
+    if (!checkVectBetween(lowerRuleCurve) || !checkVectBetween(upperRuleCurve))
+    {
+        logs.warning() << "Values for rule curve series should be between 0 and 1";
+        return false;
+    }
+
     return true;
 }
 
 bool Series::loadFromFolder(const std::string& folder)
 {
     bool ret = true;
 #define SEP Yuni::IO::Separator
-    ret = loadVector(folder + SEP + "PMAX-injection.txt", maxInjectionModulation) && ret;
-    ret = loadVector(folder + SEP + "PMAX-withdrawal.txt", maxWithdrawalModulation) && ret;
-    ret = loadVector(folder + SEP + "inflows.txt", inflows) && ret;
-    ret = loadVector(folder + SEP + "lower-rule-curve.txt", lowerRuleCurve) && ret;
-    ret = loadVector(folder + SEP + "upper-rule-curve.txt", upperRuleCurve) && ret;
+    ret = loadFile(folder + SEP + "PMAX-injection.txt", maxInjectionModulation) && ret;
+    ret = loadFile(folder + SEP + "PMAX-withdrawal.txt", maxWithdrawalModulation) && ret;
+    ret = loadFile(folder + SEP + "inflows.txt", inflows) && ret;
+    ret = loadFile(folder + SEP + "lower-rule-curve.txt", lowerRuleCurve) && ret;
+    ret = loadFile(folder + SEP + "upper-rule-curve.txt", upperRuleCurve) && ret;
 #undef SEP
     return ret;
 }
 
-bool Series::loadVector(const std::string& path, std::vector<double>& vect) const
+bool loadFile(const std::string& path, std::vector<double>& vect)
 {
-    if (!Yuni::IO::File::Exists(path))
+    logs.debug() << "Loading file " << path;
+
+    vect.reserve(HOURS_PER_YEAR);
+
+    std::ifstream file;
+    file.open(path);
+
+    if (!file)
+    {
+        logs.debug() << "File not found: " << path;
         return true;
+    }
 
-    vect.resize(HOURS_PER_YEAR);
-    return Array1DLoadFromFile(path.c_str(), vect.data(), HOURS_PER_YEAR);
+    std::string line;
+    try
+    {
+        unsigned int count = 0;
```

Comment:
```suggestion
        unsigned int line = 0;
```

Reply:
Line is already used for std::string name, I'll replace count by lineCount